rust-module/mfrc522_spi.rs

```diff
@@ -0,0 +1,79 @@
+use core::{mem, slice};
+use kernel::spi::{Spi, SpiDevice};
+use kernel::{Error, KernelResult};
+
+#[derive(Clone, Copy)]
+pub enum Mfrc522Register {
+    Version = 0x37,
+}
+
+#[derive(Debug)]
+pub enum Mfrc522Version {
+    Version1 = 0x91,
+    Version2 = 0x92,
+    NotMfrc522,
+}
+
+impl From<u8> for Mfrc522Version {
+    fn from(value: u8) -> Self {
+        match value {
+            0x91 => Self::Version1,
+            0x92 => Self::Version2,
+            _ => Self::NotMfrc522,
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+#[repr(u8)]
+enum AddressByteMode {
+    Read = 0,
+    Write = 1,
+}
+
+#[repr(packed)]
+struct AddressByte {
+    addr: Mfrc522Register,
+    mode: AddressByteMode,
+}
+
+impl AddressByte {
+    fn new(addr: Mfrc522Register, mode: AddressByteMode) -> Self {
+        AddressByte { addr, mode }
+    }
+
+    fn to_byte(&self) -> u8 {
+        (self.addr as u8 & 0b111111) << 1 | self.mode as u8 & 0b1
```

Comment:
```suggestion
        (self.addr as u8 & 0b00111111) << 1 | self.mode as u8 & 0b1
```

Reply:
I thought that only seeing ones was prone to errors as you just think ALL bits are ones.
This clearly shows there are two zero bits.